jiant/pytorch_transformers_interface/__init__.py

```diff
@@ -7,8 +7,58 @@
 WWM-BERT.
 
 This AllenNLP issue is relevant: https://github.com/allenai/allennlp/issues/3067
+
+Note: huggingface forgot to upload bert-large-uncased-whole-word-masking-finetuned-squad
+When they fix it, remove this note
+https://github.com/huggingface/pytorch-transformers/issues/763 
+
+TODO: we do not support non english xlm, if you need them, add some code in XLMEmbedderModule
+to prepare langs input to pytorch_transformers.XLMModel
 """
 
 
-def input_module_uses_pytorch_transformers(module_name):
+def input_module_support_pair_embedding(module_name):
     return module_name.startswith("bert-") or module_name.startswith("xlnet-")
+
+
+def input_module_uses_pytorch_transformers(module_name):
+    return (
+        module_name.startswith("bert-")
+        or module_name.startswith("xlnet-")
+        or module_name.startswith("gpt2")
+        or module_name.startswith("openai-gpt")
+        or module_name.startswith("transfo-xl-")
+        or module_name.startswith("xlm-")
+    )
+
+
+def input_module_tokenized_name(module_name):
+    input_module_to_pretokenized = {
+        "bert-base-uncased": "bert_uncased",
+        "bert-large-uncased": "bert_uncased",
+        "bert-large-uncased-whole-word-masking": "bert_uncased",
+        "bert-large-uncased-whole-word-masking-finetuned-squad": "bert_uncased",
+        "bert-base-cased": "bert_cased",
+        "bert-large-cased": "bert_cased",
+        "bert-large-cased-whole-word-masking": "bert_cased",
+        "bert-large-cased-whole-word-masking-finetuned-squad": "bert_cased",
+        "bert-base-cased-finetuned-mrpc": "bert_cased",
+        "bert-base-multilingual-uncased": "bert_multilingual_uncased",
+        "bert-base-multilingual-cased": "bert_multilingual_cased",
+        "bert-base-chinese": "bert_chinese",
+        "bert-base-german-cased": "bert_german_cased",
+        "xlnet-base-cased": "xlnet_cased",
+        "xlnet-large-cased": "xlnet_cased",
+        "openai-gpt": "openai_gpt",
+        "gpt2": "gpt2",
+        "gpt2-medium": "gpt2",
+        "transfo-xl-wt103": "transfo_xl",
+        "xlm-mlm-en-2048": "xlm_en",
+        "xlm-mlm-ende-1024": "xlm_ende",
+        "xlm-mlm-enfr-1024": "xlm_enfr",
+        "xlm-clm-enfr-1024": "xlm_enfr",
+        "xlm-mlm-enro-1024": "xlm_enro",
+        "xlm-mlm-tlm-xnli15-1024": "xlm_xnli",
+        "xlm-mlm-xnli15-1024": "xlm_xnli",
+    }
```

Comment:
Is there any way to build this map by importing constants from pytorch_transformers, or is the structure of that code too complex/varied for that to work?